chain/chain/src/types.rs

```diff
@@ -116,6 +117,7 @@ pub trait RuntimeAdapter: Send + Sync {
     fn filter_transactions(
         &self,
         block_index: BlockIndex,
+        block_timestamp: u64,
```

Comment:
In `validate_tx` and `filter_transactions` block_timestamp is not really needed but is required because of `ApplyState`. So does it make sense to make the `block_timestamp` field in `ApplyState` optional?

Reply:
I think it make sense in ApplyState. If we want to limit scope of what validate_tx / filter_tx have access - we should add a different structure for them.